runtime/depends_on.cc

```diff
@@ -319,4 +319,121 @@ bool is_pure(expr_ref x) {
   return v.is_pure;
 }
 
+namespace {
+
+// Find the buffers accessed in a stmt. This is trickier than it seems, we need to track the lineage of buffers and
+// report the buffer as it is visible to the caller. So something like:
+//
+//   x = crop_dim(y, ...) {
+//     call(f, {x}, {})
+//   }
+//
+// Will report that y is consumed in the stmt, not x.
+class dependency_finder : public recursive_node_visitor {
+  bool input, output;
+  
+  symbol_map<var> aliases;
+
+public:
+  std::vector<var> result;
```

Comment:
Why this is a vector? Keeping it sorted would be O(n^2) in the worst case.

Reply:
I'm trying to avoid putting `std::set`/`std::map` in the API, because I expect we won't be be able to use those at some point (on targets with weak malloc/free doing so many small allocations...).

I also think searching the result via `std::binary_search` should be a lot better than `std::set::count` or the like (same complexity, much better memory access patterns).